pkg/cmd/retrieve.go

```diff
@@ -30,9 +31,19 @@ func (s *ClientRetrieve) Customize(cmd *cobra.Command) {
 }
 
 func (s *ClientRetrieve) Run(cmd *cobra.Command, args []string) error {
-	c, err := s.getClient()
-	if err != nil {
-		return err
+
+	var err error
+	var c client.Client
+	if s.Archive != "" {
```

Comment:
Nit: maybe you can put this logic into `s.getClient` so you don't have to rewrite this in top level everywhere? so something like 
```
func (s Client) getClient() {
  if s.archive != "" {
      return clientFromArchive()
  }
  else {
     return normalClient()
}
```

Reply:
Done :heavy_check_mark: